internal/db/src/schema/budgets.ts

```diff
@@ -0,0 +1,56 @@
+import { relations } from "drizzle-orm";
+import {
+  boolean,
+  datetime,
+  float,
+  index,
+  json,
+  mysqlEnum,
+  mysqlTable,
+  varchar,
+} from "drizzle-orm/mysql-core";
+import { workspaces } from "./workspaces";
+
+export const budgets = mysqlTable(
+  "budgets",
+  {
+    id: varchar("id", { length: 256 }).primaryKey(),
+    workspaceId: varchar("workspace_id", { length: 256 })
+      .notNull()
+      .references(() => workspaces.id, { onDelete: "cascade" }),
+    name: varchar("name", { length: 256 }),
+    type: mysqlEnum("type", ["soft", "hard"]).notNull(),
+    /**
+     * Sets if the budget is enabled or not.
+     */
+    enabled: boolean("enabled").default(true).notNull(),
+    fixedAmount: float("fixed_amount").notNull(),
+    data: json("data")
+      .$type<{
+        /**
+         * Additional emails to notify.
+         */
+        additionalEmails?: string[];
+
+        /**
+         * Webhook URL to POST trigger when metered resources reached set amount.
+         * @dev TODO: Not used currently.
+         */
+        webhookUrl?: string;
```

Comment:
Just explanatory to showcase future customization of budget notification with webhookUrls, Slack, etc.